app/src/main/java/com/example/plannr/admin/AdminAddPresenter.java

```diff
@@ -0,0 +1,36 @@
+package com.example.plannr.admin;
+
+import android.widget.Button;
+
+import com.example.plannr.services.DatabaseConnection;
+import com.google.firebase.database.DatabaseReference;
+
+public class AdminAddPresenter {
+    private AdminAddView view;
+    private DatabaseConnection db = new DatabaseConnection();
```

Comment:
you should only declare the variable here, not instantiate. In the constructor is where you want to instantiate it. Also because it's a singleton class, you want to call `db = DatabaseConnection().getInstance();`